src/Generator/ParameterGenerator.php

```diff
@@ -163,14 +163,20 @@ public function __construct(
     }
 
     /**
-     * @param  string $type
+     * @param  mixed $type
      * @return ParameterGenerator
      */
     public function setType($type)
     {
-        $this->type = TypeGenerator::fromTypeString($type);
+        if (is_array($type)) {
+            $this->type = TypeGenerator::fromTypeArray($type);
+            return $this;
```

Comment:
The duplicate `return $this;` could be moved to the bottom outside the if statements